### PR TITLE
SONARIAC-1090 S6437 Hardcoded secrets in mysql, msqladmin and mysqldump

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/docker/src/test/resources/checks/SecretsGenerationCheck/sshpass.dockerfile [hunk 1]
```diff
@@ -8,6 +8,10 @@ FROM ubuntu:22.04
 
 ARG PASSWORD
 
+#tmp
+# Noncompliant@+1
+RUN sshpass "-p" "${PASSWORD}" ssh user@hostname
+
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/test/resources/checks/SecretsGenerationCheck/sshpass.dockerfile:14]: Did you forget to remove this one? I see a #tmp comment and it seems you added the same test at line 35.

File: iac-extensions/docker/src/test/resources/checks/SecretsGenerationCheck/mysql_locations.dockerfile
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/test/resources/checks/SecretsGenerationCheck/mysql_locations.dockerfile:1]: I believe I didn't saw the ones bellow, which should be compliant:
```
RUN mysql --password db_name
RUN mysqldump -p db_name
```

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsGenerationCheck.java [hunk 2]
```diff
@@ -87,17 +91,42 @@ public class SecretsGenerationCheck implements IacCheck {
...
-      (resolution.value().length() > 2 || resolution.argument().expressions().size() > 1))
-    .build();
+  private static final CommandDetector SSHPASS_P_FLAG_NO_SPACE_PWD = commandsFlagNoSpacePwd(List.of("sshpass"), "-p");
+
+  private static final List<String> MYSQL_COMMANDS = List.of("mysql", "mysqladmin", "mysqldump");
+  private static final CommandDetector MYSQL_PASSWORD_EQUALS_PWD = commandsFlagEquals(MYSQL_COMMANDS, PASSWORD_FLAG);
+
+  private static final CommandDetector MYSQL_P_FLAG_NO_SPACE_PWD = commandsFlagNoSpacePwd(MYSQL_COMMANDS, "-p");
+
+  private static CommandDetector commandsFlagNoSpacePwd(List<String> commands, String flag) {
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsGenerationCheck.java:101]: Maybe rename `commandsFlagNoSpacePwd` to `commandsFlagNoSpace`, as this method does not seem to be dedicated to password flags only.

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM! Interesting generalization with `commandsFlagEquals` and `commandsFlagNoSpacePwd`, maybe (now or later) it could prove useful in other checks.
PLease see my few comments.

File: iac-extensions/docker/src/test/resources/checks/SecretsGenerationCheck/sshpass.dockerfile (hunk 1 above)
Comment (mstachniuk) [reply iac-extensions/docker/src/test/resources/checks/SecretsGenerationCheck/sshpass.dockerfile:14]: You right, it was for debugging

File: iac-extensions/docker/src/test/resources/checks/SecretsGenerationCheck/mysql_locations.dockerfile
Comment (mstachniuk) [reply iac-extensions/docker/src/test/resources/checks/SecretsGenerationCheck/mysql_locations.dockerfile:1]: Good catch added it to `mysql_template.dockerfile`

**Pull request merged**